test/e2e/azure_test.go

```diff
@@ -142,59 +142,61 @@ var _ = Describe("Workload cluster creation", func() {
 
 	if os.Getenv("LOCAL_ONLY") != "true" {
 		Context("Creating a private cluster", func() {
-			It("Creates a public management cluster in a custom vnet", func() {
-				clusterName = getClusterName(clusterNamePrefix, "public-custom-vnet")
-				Context("Creating a custom virtual network", func() {
-					Expect(os.Setenv(AzureCustomVNetName, "custom-vnet")).NotTo(HaveOccurred())
-					additionalCleanup = SetupExistingVNet(ctx,
-						"10.0.0.0/16",
-						map[string]string{fmt.Sprintf("%s-controlplane-subnet", os.Getenv(AzureCustomVNetName)): "10.0.0.0/24"},
-						map[string]string{fmt.Sprintf("%s-node-subnet", os.Getenv(AzureCustomVNetName)): "10.0.1.0/24"},
-						fmt.Sprintf("%s-azure-bastion-subnet", os.Getenv(AzureCustomVNetName)),
-						"10.0.2.0/24",
-					)
-				})
-
-				clusterctl.ApplyClusterTemplateAndWait(ctx, clusterctl.ApplyClusterTemplateAndWaitInput{
-					ClusterProxy: bootstrapClusterProxy,
-					ConfigCluster: clusterctl.ConfigClusterInput{
-						LogFolder:                filepath.Join(artifactFolder, "clusters", bootstrapClusterProxy.GetName()),
-						ClusterctlConfigPath:     clusterctlConfigPath,
-						KubeconfigPath:           bootstrapClusterProxy.GetKubeconfigPath(),
-						InfrastructureProvider:   clusterctl.DefaultInfrastructureProvider,
-						Flavor:                   "custom-vnet",
-						Namespace:                namespace.Name,
-						ClusterName:              clusterName,
-						KubernetesVersion:        e2eConfig.GetVariable(capi_e2e.KubernetesVersion),
-						ControlPlaneMachineCount: pointer.Int64Ptr(1),
-						WorkerMachineCount:       pointer.Int64Ptr(1),
-					},
-					WaitForClusterIntervals:      e2eConfig.GetIntervals(specName, "wait-cluster"),
-					WaitForControlPlaneIntervals: e2eConfig.GetIntervals(specName, "wait-control-plane"),
-					WaitForMachineDeployments:    e2eConfig.GetIntervals(specName, "wait-worker-nodes"),
-				}, result)
+			Context("Required cluster flavor test", func() {
```

Comment:
Just a thought: would it be better to add a [REQUIRED] prefix or suffix to the higher-level context instead of adding a separate Ginkgo context? Right now it reads like this in the e2e output:

`Workload cluster creation Creating a private cluster Required cluster flavor test Creates a public management cluster in a custom vnet`

which kind of stumbles

@mboersma thoughts?

Reply:
So basically re-organize the E2E tests so that all of the required tests are children of a single parent context? (and the same for optional and experimental)

Reply:
Without reorganizing the test, we could add the [OPTIONAL] / [REQUIRED] prefix to the existing contexts. Or, having one parent context for each category of tests would also work. Essentially:

- Option 1: 

```
Workload cluster creation 
    Creating a private cluster [REQUIRED] 
        Creates a public management cluster in a custom vnet
```

- Option 2: 

```
Required workload cluster flavor test 
    Creating a private cluster 
        Creates a public management cluster in a custom vnet
```

Reply:
I slightly prefer option 2, but it would be tricker to make it backwards-compatible with the existing test-infra configs, and would be harder to review, so I'll opt for option 1 (which will be even easier to review compared to what's there right now :))

Reply:
Updated this PR, and the test-infra PR that will operate against these changes after this one merges:

https://github.com/kubernetes/test-infra/pull/25307